### ISSUE TITLE
System path fails to build when switching to a configuration

### ISSUE THREAD

**Issue by pimeys** [NixOS/nixpkgs#349816]
### Describe the bug

Switching to a configuration has been broken a couple of last commits on nixos unstable.

### Steps To Reproduce
1. `sudo nixos-rebuild switch`

Result

```
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/lvln9vwhlhwv7w4kf5yqvh452c6781bz-system-path.drv^*'
created 49831 symlinks in user environment
install-info: No such file or directory for gzip -d
```

### Expected behavior
A switch should succeed and install the new configuration.

### Additional context

Failing configuration: https://git.sr.ht/~pimeys/nixos/

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.3-cachyos, NixOS, 24.11 (Vicuna), 24.11.20241016.5785b6b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/w7nfn4lcfisddz14b0y8n05rjf0dl5lq-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Artturin** [comment]
Test if it's from https://search.nixos.org/options?channel=unstable&show=documentation.enable&from=0&size=50&sort=relevance&type=packages&query=documentation.enable

https://github.com/NixOS/nixpkgs/blob/0065f778752685fca1be1df542ff13ddc42bd32e/nixos/modules/misc/documentation.nix#L363

**Comment by pimeys** [comment]
@Artturin Yep. Without manpages both of my systems build successfully. I can live with this for now, but I'm sure going to miss my `man configuration.nix`...

**Comment by Artturin** [comment]
Debug by adding 

`set -x`
`set +x`
Above and below those lines

**Comment by pimeys** [comment]
I'm not sure what happened, but it seems to build right now. What I did is I fixed another issue in here:

https://github.com/NixOS/nixpkgs/pull/350030

And the docs seem to build correctly.
